Test failures regarding mpio

To assist reproducing bugs, please include the following:
 * Operating System: Fedora rawhide
 * Python version: 3.9.0
 * Where Python was acquired: system Python
 * h5py version: latest master
 * HDF5 version: 1.10.7

```
+ /usr/bin/python3 setup.py test
running test
running build_py
running build_ext
Loading library to get version: libhdf5.so
********************************************************************************
                       Summary of the h5py configuration

HDF5 include dirs: []
HDF5 library dirs: []
     HDF5 Version: (1, 10, 7)
      MPI Enabled: False
 Rebuild Required: False

********************************************************************************
Executing cythonize()
============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.0.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/orion/fedora/h5py/h5py-2.10.0/serial, configfile: pytest.ini
plugins: cov-2.10.1, arraydiff-0.3, doctestplus-0.7.0, pyfakefs-3.5.8, remotedata-0.3.2, openfiles-0.5.0, mock-3.3.1, xdist-2.1.0, timeout-1.4.2, forked-1.3.0
collected 651 items
...
==================================== ERRORS ====================================
_____________________ ERROR at setup of TestMPI.test_mpio ______________________
file /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py, line 725
      def test_mpio(self, mpi_file_name):
E       fixture 'mpi_file_name' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, class_mocker, cov, doctest_namespace, fs, mocker, module_mocker, monkeypatch, no_cover, package_mocker, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, session_mocker, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id, writable_file
>       use 'pytest --fixtures [testpath]' for help on them.

/home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py:725
__________________ ERROR at setup of TestMPI.test_mpi_atomic ___________________
file /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py, line 733
      @pytest.mark.skipif(h5py.version.hdf5_version_tuple < (1, 8, 9),
          reason="mpio atomic file operations were added in HDF5 1.8.9+")
      def test_mpi_atomic(self, mpi_file_name):
E       fixture 'mpi_file_name' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, class_mocker, cov, doctest_namespace, fs, mocker, module_mocker, monkeypatch, no_cover, package_mocker, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, session_mocker, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id, writable_file
>       use 'pytest --fixtures [testpath]' for help on them.

/home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py:733
__________ ERROR at setup of TestMPI.test_close_multiple_mpio_driver ___________
file /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py, line 744
      def test_close_multiple_mpio_driver(self, mpi_file_name):
E       fixture 'mpi_file_name' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, class_mocker, cov, doctest_namespace, fs, mocker, module_mocker, monkeypatch, no_cover, package_mocker, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, session_mocker, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id, writable_file
>       use 'pytest --fixtures [testpath]' for help on them.

/home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py:744
=============================== warnings summary ===============================
h5py/tests/test_file.py:723
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_file.py:723: PytestUnknownMarkWarning: Unknown pytest.mark.mpi - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi

h5py/tests/test_filters.py:83
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_filters.py:83: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

h5py/tests/test_h5pl.py:23
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_h5pl.py:23: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

h5py/tests/test_h5pl.py:31
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_h5pl.py:31: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

h5py/tests/test_h5pl.py:41
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_h5pl.py:41: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

h5py/tests/test_h5pl.py:51
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_h5pl.py:51: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

h5py/tests/test_h5pl.py:61
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_h5pl.py:61: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

h5py/tests/test_h5pl.py:70
  /home/orion/fedora/h5py/h5py-2.10.0/serial/build/lib.linux-x86_64-3.9/h5py/tests/test_h5pl.py:70: PytestUnknownMarkWarning: Unknown pytest.mark.mpi_skip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.mpi_skip

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
ERROR h5py/tests/test_file.py::TestMPI::test_mpio
ERROR h5py/tests/test_file.py::TestMPI::test_mpi_atomic
ERROR h5py/tests/test_file.py::TestMPI::test_close_multiple_mpio_driver
============ 634 passed, 14 skipped, 8 warnings, 3 errors in 12.92s ============
```

--- a/.authors.yml
+++ b/.authors.yml
@@ -166,6 +166,8 @@
   num_commits: 4
   first_commit: 2019-07-02 14:30:12
   github: Anthchirp
+  alternate_emails:
+  - 2102431+Anthchirp@users.noreply.github.com
 - name: root
   email: root@acdlab11.lims.acdlabs.ru
   num_commits: 1

--- a/news/require-pytest-mpi.rst
+++ b/news/require-pytest-mpi.rst
@@ -1,0 +1,29 @@
+New features
+------------
+
+* <news item>
+
+Deprecations
+------------
+
+* <news item>
+
+Exposing HDF5 functions
+-----------------------
+
+* <news item>
+
+Bug fixes
+---------
+
+* <news item>
+
+Building h5py
+-------------
+
+* <news item>
+
+Development
+-----------
+
+* If pytest is missing pytest-mpi then it will now fail immediately with a clear warning message.

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,3 +1,4 @@
 [pytest]
 filterwarnings =
     error::h5py.h5py_warnings.H5pyDeprecationWarning
+required_plugins = pytest-mpi

